packages/server/src/fhir/operations/execute.ts

```diff
@@ -39,7 +39,12 @@ export interface BotExecutionResult {
 export const executeHandler = asyncWrap(async (req: Request, res: Response) => {
   const { id } = req.params;
   const repo = res.locals.repo as Repository;
-  const bot = await repo.readResource<Bot>('Bot', id);
+
+  // First read the bot as the user to verify access
+  await repo.readResource<Bot>('Bot', id);
+
+  // Then read the bot as system user to load extended metadata
+  const bot = await systemRepo.readResource<Bot>('Bot', id);
```

Comment:
This seems a little weird to me, that you would need to do a 2nd read with the systemRepo. Why do we need the extended properties on this call? And could we have the `repo.readResource` call return the extend metada (maybe with an additional param)?

My concern is that using the systemRepo is a more privileged accessor. It seems fine in this use case, but not sure if it could hurt us later

Reply:
The problem is that the user might not have access to necessary metadata.

The first call reads the bot as the user calling the endpoint.  This verifies access to the bot.

The second call reads the bot as system to ensure that all metadata is available (mostly project metadata).

We could potentially refactor to expose "readResourceImpl", which reads the resource without stripping hidden fields.

Reply:
Yeah I see. Maybe the better refactor would be to have an explicit `repo.checkAccess` method, rather than using `readResource` as a proxy?

Reply:
I agree in principle.  We need that feature in a few areas.

At present, our most robust permission check is implemented in SQL.  A `User`'s `AccessPolicy` can define `criteria` as a list of search parameters.  We need to implement that in SQL for basic querying.

It would be nice to have a complete parallel implementation for `checkAccess` completely in memory.  We're most of the way there, but to use it for security purposes we need to be at 100% parity for all of the long tail search features like `system|value` token searches, etc.

You may remember this conversation from when we first implemented Redis caching.  We only use the Redis cache for users who don't have an `AccessPolicy` for this same reason -- we would need a `checkAccess` to verify access in memory without using SQL.

So, in the meantime, using `readResource` and/or `readResourceImpl` are the most reliable methods.

I suppose if it's just for readability purposes, we could have a `repo.checkAccess()` that calls `readResource()`, but it would just be for optics.

Reply:
Yeah that makes sense. Don't think it's worth blocking this PR, but I'll file a ticket for `repo.checkAccess()`